st2common/st2common/services/triggers.py

```diff
@@ -101,13 +101,22 @@ def get_trigger_type_db(ref):
 
 def _get_trigger_api_given_rule(rule):
     trigger = rule.trigger
-    triggertype_ref = ResourceReference.from_string_reference(trigger.get('type'))
     trigger_dict = {}
-
-    trigger_dict['name'] = triggertype_ref.name
-    trigger_dict['pack'] = triggertype_ref.pack
+    triggertype_ref = ResourceReference.from_string_reference(trigger.get('type'))
+    trigger_dict['pack'] = trigger_dict.get('pack', triggertype_ref.pack)
     trigger_dict['type'] = triggertype_ref.ref
-    trigger_dict['parameters'] = rule.trigger.get('parameters', {})
+
+    trigger_ref_str = trigger.get('ref', None)
```

Comment:
I believe this breaks existing rules which refer to a trigger using `type` instead of `ref` argument. We need to make sure `name` gets set even if `ref` is not provided (we can infer name from the `type`). If name is not set, we will generate a random one, which will cause a miss-match in the rules engine...

All of our examples and documentation use `type` and not `ref` in the trigger part of the rule definition (see `contrib/examples/rules/`, blog posts, documentation, etc.).

I was tracking down this weird behavior for quite a while, and just recently pushed a fix - https://github.com/StackStorm/st2/commit/c7b0a68d0eb2e295e645d3a7edd1d0da45533480. This problem is that this issue only happens if a rule is created before a particular trigger type is registered.

Anyway, I will test it and dig more into it. We need to make sure this works correctly because as mentioned above, rules and triggers were broken if rule was created before the trigger being registered until I pushed the fix above.